x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/watcher/PutWatchRequest.java

```diff
@@ -129,7 +140,7 @@ public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = null;
         if (id == null) {
             validationException = ValidateActions.addValidationError("watch id is missing", validationException);
-        } else if (WatcherUtils.isValidId(id) == false) {
+        } else if (isValidId(id) == false) {
```

Comment:
does "isValidId" do more than just check whitespace? if not should we rename it? it seems odd that you can check for a valid id and know that the case for failure was whitespace here, given that its a blackbox checking for whatever we define as "validity"

Reply:
see below, this also checks for `!= null` and non-emptiness...

Reply:
can we clean up the validation string to say "or is empty" so that passing `""` in does not validate to "sorry your watch ID contains whitespace"

Reply:
I'd do that in a follow up PR